easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1152,6 +1153,52 @@ def _validate(self, attr, values):  # private method
         if self[attr] and self[attr] not in values:
             raise EasyBuildError("%s provided '%s' is not valid: %s", attr, self[attr], values)
 
+    def handle_external_module_metadata_by_probing_modules(self, dep_name):
+        """
+        helper function for handle_external_module_metadata
+        handles metadata for external module dependencies when there is not entry in the
+        metadata file
+
+        It should look for the pair of variables definitions in the available modules
+          1. CRAY_XXXX_PREFIX and CRAY_XXXX_VERSION
+          2. CRAY_XXXX_DIR and CRAY_XXXX_VERSION
+          2. CRAY_XXXX_ROOT and CRAY_XXXX_VERSION
+          5. XXXX_PREFIX and XXXX_VERSION
+          4. XXXX_DIR and XXXX_VERSION
+          5. XXXX_ROOT and XXXX_VERSION
+          3. XXXX_HOME and XXXX_VERSION
+
+        If neither of the pairs is found, then an empty dictionary is returned
+        """
+        dependency = {}
+
+        short_ext_modname = dep_name.split('/')[0]
+        short_ext_modname_upper = convert_name(short_ext_modname, upper=True)
+
+        allowed_pairs = [
+            ('CRAY_%s_PREFIX' % short_ext_modname_upper, 'CRAY_%s_VERSION' % short_ext_modname_upper),
+            ('CRAY_%s_DIR' % short_ext_modname_upper, 'CRAY_%s_VERSION' % short_ext_modname_upper),
+            ('CRAY_%s_ROOT' % short_ext_modname_upper, 'CRAY_%s_VERSION' % short_ext_modname_upper),
+            ('%s_PREFIX' % short_ext_modname_upper, '%s_VERSION' % short_ext_modname_upper),
+            ('%s_DIR' % short_ext_modname_upper, '%s_VERSION' % short_ext_modname_upper),
+            ('%s_ROOT' % short_ext_modname_upper, '%s_VERSION' % short_ext_modname_upper),
+            ('%s_HOME' % short_ext_modname_upper, '%s_VERSION' % short_ext_modname_upper),
+        ]
+
+        for p, v in allowed_pairs:
+            prefix = self.modules_tool.get_variable_from_modulefile(dep_name, p)
+            version = self.modules_tool.get_variable_from_modulefile(dep_name, v)
+
+            if prefix and version:
```

Comment:
if prefix is found but not version, and if the version is in `dep_name`, can't we simply use that one?

Reply:
The point is to only trust the metadata obtained by probing the module if it provides both the installation prefix and software version.
If it only provides one of both, we could have a "lucky hit", and the obtained metadata may be less trustworthy...

Let's go forward as is with this strict requirement, which can easily be loosened up later (the other way around is more difficult).

The other scenario, where the prefix is already known from the metadata file, but (only) the version is available through the module, may actually be more likely (since the prefix can be specified as the name of the environment variable specifying the installation prefix, not an actual hardcoded prefix, which can be done generically in the metadata file).